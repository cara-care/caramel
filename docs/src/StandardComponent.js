import React from 'react';
import './assets/css/standardcomponent.css';

const StandardComponent = () => {
  return (
    <div>
      <h2>Accordion</h2>
      <div className="imageTableContainer">
        <div>
          <img style={{maxWidth: "300px", margin: "24px"}} src={require("./assets/images/Accordion.gif")} />
        </div>
        <div className="divTable">
          <div className="divTableBody">
            <div className="divTableRow divTableHeading">
              <div className="divTableCell">Property</div>
              <div className="divTableCell">Type</div>
              <div className="divTableCell">Default</div>
              <div className="divTableCell">Description</div>
            </div>
            <div className="divTableRow">
              <div className="divTableCell">
                <pre className="codeArea">
                  {`
list: {
  image ?: ImageURISource;
  name: string;
  description: string;
  iconStyle?: ImageStyle;
  titleStyle?: TextStyle;
  descriptionStyle?: TextStyle;
  arrowStyle?: ImageStyle;
  separatorStyle?: ViewStyle;
  rowContainerStyle?: ViewStyle;
  onOpen?: () => void;
  onClose?: () => void;
}[]
                  `}
                </pre>
              </div>
              <div className="divTableCell">Array</div>
              <div className="divTableCell">-</div>
              <div className="divTableCell">The list array to render the AccordionRow components. You can add events and
            styles per row.</div>
            </div>
            <div className="divTableRow">
              <div className="divTableCell">
                style
          </div>
              <div className="divTableCell">ViewStyle |Â undefined</div>
              <div className="divTableCell">-</div>
              <div className="divTableCell">Style for the FlatList parent</div>
            </div>
            <div className="divTableRow">
              <div className="divTableCell">
                disableAutoClose
          </div>
              <div className="divTableCell">boolean | undefined</div>
              <div className="divTableCell">false</div>
              <div className="divTableCell">Disables the rows closing themselves automatically when another row is
            opened within the same Accordion.</div>
            </div>
            <div className="divTableRow">
              <div className="divTableCell">
                animate
          </div>
              <div className="divTableCell">boolean | undefined</div>
              <div className="divTableCell">false</div>
              <div className="divTableCell">Animates the opening and closing events of the rows.</div>
            </div>
          </div>
        </div>
      </div>
      <h4>Example</h4>
      <pre className="codeArea">
        {
`<Accordion
  list={[
      {
        image: require('./caramel.png'),
        name: 'Accordion1',
        description: 'Lorem ipsum dolor sit amet.',
      },
      {
        image: require('./caramel.png'),
        name: 'Accordion2',
        description: 'Lorem ipsum dolor sit amet.',
      },
    ]}
  animate />`
  }
  </pre>
    </div>
  );
}

export default StandardComponent;
