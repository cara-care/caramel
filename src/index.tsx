import Text from './components/Text';
import ColorText from './components/ColorText';
import AnimatedProgressBar from './components/AnimatedProgressBar';
import ThermoScore from './components/ThermoScore';
import AccordionRow from './organisms/AccordionRow';
import Slider from './components/Slider';
import BottomActionSheet from './components/BottomActionSheet';
import Accordion from './components/Accordion';
import Container from './components/Container';
import IconButton from './components/IconButton';
import CoverScrollView from './components/CoverScrollView';
import theme from './utils/Theme';

export {
  default as TouchableBreath,
  TouchableBreathProps,
} from './components/TouchableBreath';
export {
  Text,
  ColorText,
  AnimatedProgressBar,
  Slider,
  BottomActionSheet,
  Container,
  ThermoScore,
  AccordionRow,
  Accordion,
  IconButton,
  CoverScrollView,
  theme,
};
