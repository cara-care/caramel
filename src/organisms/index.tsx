import BottomSheetAndroid from './BottomSheetAndroid';
import AccordionRow from './AccordionRow';
import CoverImage from './CoverImage';

export {BottomSheetAndroid, AccordionRow, CoverImage};
