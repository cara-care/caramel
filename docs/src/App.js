import React from 'react';
import StandardComponent from './StandardComponent';

const App = () => {
  return (
    <div>
      <StandardComponent />
    </div>
  );
}

export default App;
