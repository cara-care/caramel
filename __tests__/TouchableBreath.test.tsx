import 'react-native';
import React from 'react';
import {Text} from 'react-native';
import renderer from 'react-test-renderer';
import {TouchableBreath} from '../src/';

describe('<TouchableBreath />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <TouchableBreath>
          <Text>TouchableBreath</Text>
        </TouchableBreath>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
