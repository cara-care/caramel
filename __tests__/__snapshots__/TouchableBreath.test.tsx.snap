// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TouchableBreath /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <Text>
      TouchableBreath
    </Text>
  </View>
</View>
`;
